src/swupd.h

```diff
@@ -278,8 +279,8 @@ static inline void account_delta_miss(void)
 
 extern void print_statistics(int version1, int version2);
 
-extern int download_fullfiles(struct list *files, int *num_downloads);
-extern int download_subscribed_packs(struct list *subs, struct manifest *mom, bool required);
+extern int download_fullfiles(struct list *files, int *num_downloads, struct step step);
```

Comment:
we need a better way to handle this step struct. I don't think that the the fullfiles.c or packs.c should care about the step we are in to.

Reply:
I think this would be solved by implementing your suggestions about the `json_formatter_set_step()` and `json_formatter_get_step()`.